### ISSUE TITLE
[BUG] Merging string index with string column throws error

### ISSUE THREAD

**Issue by ayushdg** [rapidsai/cudf#1805]
**Describe the bug**
Merging two data frames, one with a string index and another one with string column throws an error. (When merging on index with one Dataframe and on column with the other Dataframe.

**Steps/Code to reproduce bug**
```
import cudf

df = cudf.DataFrame()
df['a'] = ['a','b']*5
df['b'] = ['i','b','i','a','e','i','i','i','i','i']
grouped = df.groupby(by=['a'],as_index=True).agg({'b':'min'})

df2 = cudf.DataFrame()
df2['a'] = ['c','d','e','a','b']
df2['res'] = ['a','c','d','b','e']
grouped2 = df2.groupby(by=['a'],as_index=False).max()
grouped.merge(grouped2,left_index=True,right_on='a',how='left').to_pandas()
```
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-16-f35b62b29460> in <module>
     10 df2['res'] = ['a','c','d','b','e']
     11 grouped2 = df2.groupby(by=['a'],as_index=False).max()
---> 12 grouped.merge(grouped2,left_index=True,right_on='a',how='left').to_pandas()

/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.1+471.ga66db9e.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/dataframe.py in merge(self, right, on, how, left_on, right_on, left_index, right_index, lsuffix, rsuffix, type, method, indicator, suffixes)
   1645             new_index = Series(rhs.index,
   1646                                index=RangeIndex(0, len(rhs[right_on[0]])))
-> 1647             indexed = rhs[right_on[0]][df[right_on[0]]-1]
   1648             new_index = new_index[indexed-1]
   1649             df.index = new_index

/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.1+471.ga66db9e.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/series.py in __sub__(self, other)
    460 
    461     def __sub__(self, other):
--> 462         return self._binaryop(other, 'sub')
    463 
    464     def __rsub__(self, other):

/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.1+471.ga66db9e.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/series.py in _binaryop(self, other, fn)
    420             return other._binaryop(self, fn)
    421         nvtx_range_push("CUDF_BINARY_OP", "orange")
--> 422         other = self._normalize_binop_value(other)
    423         outcol = self._column.binary_operator(fn, other._column)
    424         result = self._copy_construct(data=outcol)

/conda/envs/rapids/lib/python3.7/site-packages/cudf-0.7.1+471.ga66db9e.dirty-py3.7-linux-x86_64.egg/cudf/dataframe/series.py in _normalize_binop_value(self, other)
    536             return Series(other)
    537         else:
--> 538             col = self._column.normalize_binop_value(other)
    539             return self._copy_construct(data=col)
    540 

AttributeError: 'StringColumn' object has no attribute 'normalize_binop_value'
```

**Expected behavior**
A normal dataframe merge operation.

**Environment details (please complete the following information):**
 - Environment location: Docker
 - Method of cuDF install: from source 0.8
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used
 - Please run and attach the output of the `cudf/print_env.sh` script to gather relevant environment details: On 0.8 at commit a66db9e9fa56328ab24afcc23cc1a117a0cdbf0a
 

**Additional context**
Merge works if both are string columns and not string index. 
```
import cudf

df = cudf.DataFrame()
df['a'] = ['a','b']*5
df['b'] = ['i','b','i','a','e','i','i','i','i','i']
grouped = df.groupby(by=['a'],as_index=False).agg({'b':'min'})

df2 = cudf.DataFrame()
df2['a'] = ['c','d','e','a','b']
df2['res'] = ['a','c','d','b','e']
grouped2 = df2.groupby(by=['a'],as_index=False).max()
grouped.merge(grouped2,left_on='a',right_on='a',how='left').to_pandas()
```

Merge works with integer index on integer column
```
import cudf

df = cudf.DataFrame()
df['a'] = [1,2]*5
df['b'] = ['i','b','i','a','e','i','i','i','i','i']
grouped = df.groupby(by=['a'],as_index=True).agg({'b':'min'})

df2 = cudf.DataFrame()
df2['a'] = [3,4,5,1,2]
df2['res'] = ['a','c','d','b','e']
grouped2 = df2.groupby(by=['a'],as_index=False).max()
grouped.merge(grouped2,left_index=True,right_on='a',how='left').to_pandas()
```

Merge works when both are string index
```
import cudf

df = cudf.DataFrame()
df['a'] = ['a','b']*5
df['b'] = ['i','b','i','a','e','i','i','i','i','i']
grouped = df.groupby(by=['a'],as_index=True).agg({'b':'min'})

df2 = cudf.DataFrame()
df2['a'] = ['c','d','e','a','b']
df2['res'] = ['a','c','d','b','e']
grouped2 = df2.groupby(by=['a'],as_index=True).max()
grouped.merge(grouped2,left_index=True,right_index=True,how='left').to_pandas()
```

**Comment by thomcom** [comment]
Great analysis @ayushdg thank you!

**Comment by thomcom** [comment]
This appears to be fixed by #1850. I'll update more thoroughly when it is ready to merge.

**Comment by kkraus14** [comment]
Fixed by #1850, if there's still a problem please open a new issue
